optaplanner-core/src/main/java/org/optaplanner/core/api/score/stream/ConstraintCollectors.java

```diff
@@ -526,36 +526,380 @@ private static Runnable innerCountDistinctLong(CountDistinctLongResultContainer
     // min
     // ************************************************************************
 
+    /**
+     * Returns a collector that finds a minimum value in a group of {@link Comparable} elements.
+     * <p>
+     *     It is expected that the {@link Comparable}'s {@link Comparable#compareTo(Object)} is consistent with
+     *     {@link #equals(Object)}.
+     *     In other words, if two elements compare to equal, any of them may be returned by the collector as they are
+     *     considered to be {@link #equals(Object)}.
```

Comment:
I would change this paragraph into something like this:

```
Important: The {@link Comparable}'s {@link Comparable#compareTo(Object)} must be <i>consistent with equals</i>, such that <tt>e1.compareTo(e2) == 0</tt> has the same boolean value as <tt>e1.equals(e2)</tt>.
In other words, if two elements compare to zero, any of them can be returned by the collector. It can even differ between 2 score calculations on the exact same {@link PlanningSolution} state, due to incremental score calculation.
```

Motivation:
- start with "important" to stand out (we do this in a few other javadocs too).
-  Use "must" instead of "It is expected that ..." because that's more direct, stronger (which is appropriate here) and clearer.
- Mimic javadocs of Comparable on OpenJDK, use same terminology and phrasing.